squire_sdk/src/server/gathering/hall.rs

```diff
@@ -181,8 +181,15 @@ impl GatheringHall {
     }
 }
 
-impl Trackable<TournamentId, ()> for GatheringHallMessage {
-    fn track(msg: TournamentId, send: tokio::sync::oneshot::Sender<()>) -> Self {
-        Self::DestroyGathering(msg, send)
+// impl Trackable<TournamentId, ()> for GatheringHallMessage {
+//     fn track(msg: TournamentId, send: tokio::sync::oneshot::Sender<()>) -> Self {
+//         Self::DestroyGathering(msg, send)
+//     }
+// }
+
+impl From<(TournamentId, tokio::sync::oneshot::Sender<()>)> for GatheringHallMessage {
```

Comment:
I have been using the `derive_more` crate. It has a really useful `From` derive macro that which greatly reduces the boilerplate you need. It was actually a major reason that I got rid of the `Trackable` trait.

Reply:
I took a brief look at your code but couldn't figure out how to use it yet. I will take a look at the crate's docs later, thanks.